RZTouchID/RZTouchID.m

```diff
@@ -184,47 +200,138 @@ - (void)deletePasswordWithIdentifier:(NSString *)identifier completion:(RZTouchI
 
 #pragma mark - private methods
 
+- (void)queryPasswordWithIdentifier:(NSString *)identifier withPrompt:(NSString *)prompt completion:(RZTouchIDCompletion)completion
+{
+    CFErrorRef error = NULL;
+    SecAccessControlRef accessObject;
+    
+    accessObject = SecAccessControlCreateWithFlags(kCFAllocatorDefault, kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly, kSecAccessControlUserPresence, &error);
+    
+    //Check if password exists
+    NSDictionary *query = nil;
+    if ( self.touchIDMode == RZTouchIDModeBiometricKeychain ) {
+        query = @{
+                  (__bridge id)kSecClass: (__bridge id)kSecClassGenericPassword,
+                  (__bridge id)kSecAttrService: [self serviceNameForIdentifier:identifier],
+                  (__bridge id)kSecReturnData: @YES,
+                  (__bridge id)kSecUseOperationPrompt: prompt,
+                  (__bridge id)kSecAttrAccessControl: (__bridge id)accessObject
+                  };
+    }
+    else {
+        query = @{
+                  (__bridge id)kSecClass: (__bridge id)kSecClassGenericPassword,
+                  (__bridge id)kSecAttrService: [self serviceNameForIdentifier:identifier],
+                  (__bridge id)kSecReturnData: @YES,
+                  (__bridge id)kSecUseOperationPrompt: prompt
+                  };
```

Comment:
Same comment as above -- create mutable dict and add the 4th param only in the biometric case